Lib/asyncio/tasks.py

```diff
@@ -972,10 +983,16 @@ def _inner_done_callback(inner, cur_task=cur_task):
             else:
                 outer.set_result(inner.result())
 
-
     def _outer_done_callback(outer):
         if not inner.done():
             inner.remove_done_callback(_inner_done_callback)
+            # Keep only one callback to log on cancel
+            inner.remove_done_callback(_log_on_cancel_callback)
+            inner.add_done_callback(_log_on_cancel_callback)
+
+    if cur_task is not None:
+        inner.add_done_callback(_clear_awaited_by_callback)
```

Comment:
Now we're only adding the callback to `future_discard_from_awaited_by` when there *is* a current task. Nice.